source/extensions/filters/network/ssh/service_userauth.cc

```diff
@@ -265,15 +276,36 @@ absl::Status UpstreamUserAuthService::requestService() {
   return transport_.sendMessageToConnection(std::move(req)).status();
 }
 
+key_params_t UpstreamUserAuthService::getUpstreamKeyParams() {
+  auto key_params = DefaultUpstreamKeyParams;
+
+  auto ext_info = transport_.peerExtInfo();
+  if (ext_info) {
+    auto server_sig_algs = ext_info->getExtension<wire::ServerSigAlgsExtension>();
+    if (server_sig_algs) {
+      auto v = *server_sig_algs->public_key_algorithms_accepted;
+      absl::flat_hash_set<std::string> server_algs(v.begin(), v.end());
+      for (const auto& p : SupportedUpstreamKeyParams) {
```

Comment:
Do we need to check for the non-cert algorithm names? here is what I get from openssh by default:
```
server-sig-algs=<ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,sk-ssh-ed25519@openssh.com,sk-ecdsa-sha2-nistp256@openssh.com,rsa-sha2-512,rsa-sha2-256>
```